pkg/sql/sequence.go

```diff
@@ -155,41 +155,54 @@ func (p *planner) incrementSequenceUsingCache(
 		cacheSize = seqOpts.EffectiveCacheSize()
 	}
 
-	fetchNextValues := func() (currentValue, incrementAmount, sizeOfCache int64, err error) {
+	fetchNextValues := func() (nextValue, incrementAmount, sizeOfCache int64, err error) {
 		seqValueKey := p.ExecCfg().Codec.SequenceKey(uint32(sequenceID))
-
-		// The planner txn is only used if the sequence is accessed in the same
-		// transaction that it was created. Otherwise, we *do not* use the planner
-		// txn here, since nextval does not respect transaction boundaries.
-		// This matches the specification at
-		// https://www.postgresql.org/docs/14/functions-sequence.html.
+		sizeOfCache = cacheSize
+		incrementAmount = seqOpts.Increment
+		var res kv.KeyValue
+		var currentValue int64
 		var endValue int64
+		var expValue []byte
+
+		var getSequenceValueFunc func() (kv.KeyValue, error)
+		var cputSequenceValueFunc func() error
 		if createdInCurrentTxn {
-			var res kv.KeyValue
-			res, err = p.txn.Inc(ctx, seqValueKey, seqOpts.Increment*cacheSize)
-			endValue = res.ValueInt()
+			// The planner txn is only used if the sequence is accessed in the same
+			// transaction that it was created. Otherwise, we *do not* use the planner
+			// txn here, since nextval does not respect transaction boundaries.
+			// This matches the specification at
+			// https://www.postgresql.org/docs/14/functions-sequence.html.
+			getSequenceValueFunc = func() (kv.KeyValue, error) {
+				return p.txn.GetForUpdate(ctx, seqValueKey, kvpb.BestEffort)
+			}
+			cputSequenceValueFunc = func() error {
+				return p.txn.CPut(ctx, seqValueKey, endValue, expValue)
+			}
 		} else {
-			endValue, err = kv.IncrementValRetryable(
-				ctx, p.ExecCfg().DB, seqValueKey, seqOpts.Increment*cacheSize)
+			getSequenceValueFunc = func() (kv.KeyValue, error) {
```

Comment:
ref: https://github.com/cockroachdb/cockroach/commit/74f5728dfc354f0324f51fe271bd36a6da2c3918#diff-a726a06120b0f3619c900fa08a35968aba1b22511c7cb8ec767f14cf83d82acdR605

Reply:
ref: https://github.com/cockroachdb/cockroach/commit/5d92dd76a2867befa79b28478cbe65cd77ed8d63#diff-f91c65fd74b24454e2d9b4a54eab28c8c368b929130da23e8a2ed9d4b56d3d7c

Reply:
ref: https://github.com/cockroachdb/cockroach/commit/29f928f675d4eb79a6b3f100fd49a556971bf8bd

Reply:
I am looking for reasons that `IncrementValRetryable` is needed. It was initally introduced in 2017 so looking if it is still needed and currently not find any test failures by using only `Inc` without retry.